runhouse/resources/hardware/launcher_utils.py

```diff
@@ -122,107 +127,119 @@ def up(self, verbose: bool = True):
                 # TODO: update once secrets are updated to include pub + private key values (should have only one field)
                 "ssh_creds": "/mkandler/aws-cpu-matt-ssh-secret",
                 "sky_creds": sky_secret.rns_address,
             },
-            "force": self.force,
+            "force": force,
         }
 
         if verbose:
-            data = self.run_verbose(base_url=self.LAUNCH_URL, payload=payload)
+            data = cls.run_verbose(base_url=cls.LAUNCH_URL, payload=payload)
             logger.info("Successfully launched cluster.")
-            return data
+            cls._update_from_den_response(cluster=cluster, config=data)
+            return
 
         # Blocking call with no streaming
         resp = requests.post(
-            self.LAUNCH_URL,
+            cls.LAUNCH_URL,
             json=payload,
             headers=rns_client.request_headers(),
         )
         if resp.status_code != 200:
             raise Exception(
-                f"Received [{resp.status_code}] from Den POST '{self.LAUNCH_URL}': Failed to "
+                f"Received [{resp.status_code}] from Den POST '{cls.LAUNCH_URL}': Failed to "
                 f"launch cluster: {load_resp_content(resp)}"
             )
         data = resp.json()
         logger.info("Successfully launched cluster.")
-        return data
+        cls._update_from_den_response(data)
 
-    def teardown(self, verbose: bool = True):
+    @classmethod
+    def teardown(cls, cluster, verbose: bool = True):
         """Tearing down a cluster via Den."""
         # TODO [MK] what payload do we need here?
         if verbose:
-            self.run_verbose(base_url=self.TEARDOWN_URL)
+            cls.run_verbose(base_url=cls.TEARDOWN_URL)
             return
 
         # Run blocking call, with no streaming
         resp = requests.post(
-            self.TEARDOWN_URL,
-            json=self.cluster.config(),
+            cls.TEARDOWN_URL,
+            json=cluster.config(),
             headers=rns_client.request_headers(),
         )
         if resp.status_code != 200:
             raise Exception(
-                f"Received [{resp.status_code}] from Den POST '{self.TEARDOWN_URL}': Failed to "
+                f"Received [{resp.status_code}] from Den POST '{cls.TEARDOWN_URL}': Failed to "
                 f"teardown cluster: {load_resp_content(resp)}"
             )
 
 
 class LocalLauncher(Launcher):
     """Launcher APIs for operations handled locally via Sky."""
 
-    def __init__(self, cluster):
-        super().__init__(cluster)
-        self._validate_provider()
-
-    def _validate_provider(self):
+    @staticmethod
+    def _validate_provider(cluster):
         """Check if LocalLauncher supports the provided cloud provider."""
-        supported_providers = ["cheapest"] + self.supported_providers()
-        if self.cluster.provider not in supported_providers:
+        supported_providers = ["cheapest"] + supported_providers()
```

Comment:
prob makes sense for this to be a class method 

```suggestion
        supported_providers = ["cheapest"] + cls.supported_providers()
```